scope3_methodology/ad_tech_platform/model.py

```diff
@@ -0,0 +1,382 @@
+            bid_request_size_in_bytes
+            if bid_request_size_in_bytes
+            else defaults.bid_request_size_in_bytes
+        )
+        self.server_to_server_emissions_g_co2e_per_gb = (
+            server_to_server_emissions_g_co2e_per_gb
+            if server_to_server_emissions_g_co2e_per_gb
+            else defaults.server_to_server_emissions_g_co2e_per_gb
+        )
+        self.server_emissions_mt_co2e_per_month = (
+            server_emissions_mt_co2e_per_month
+            if server_emissions_mt_co2e_per_month
+            else defaults.server_emissions_mt_co2e_per_month
+        )
+        self.servers_processing_bid_requests_percentage = (
+            servers_processing_bid_requests_percentage
+            if servers_processing_bid_requests_percentage
+            else defaults.servers_processing_bid_requests_percentage
+        )
+        self.cookie_syncs_processed_per_bid_request = (
+            cookie_syncs_processed_per_bid_request
+            if cookie_syncs_processed_per_bid_request
+            else defaults.cookie_syncs_processed_per_bid_request
+        )
+        self.datacenter_water_intensity_h2o_m_3_per_mwh = (
+            datacenter_water_intensity_h2o_m_3_per_mwh
+            if datacenter_water_intensity_h2o_m_3_per_mwh
+            else defaults.datacenter_water_intensity_h2o_m_3_per_mwh
+        )
+        self.server_emissions_g_co2e_per_kwh = (
+            server_emissions_g_co2e_per_kwh
+            if server_emissions_g_co2e_per_kwh
+            else defaults.server_emissions_g_co2e_per_kwh
+        )
+        self.servers_processing_cookie_syncs_percentage = (
+            servers_processing_cookie_syncs_percentage
+            if servers_processing_cookie_syncs_percentage
+            else defaults.servers_processing_cookie_syncs_percentage
+        )
+        self.cookie_sync_distribution_ratio = (
+            cookie_sync_distribution_ratio
+            if cookie_sync_distribution_ratio
+            else defaults.cookie_sync_distribution_ratio
+        )
+
+        # Values without fallbacks
+        self.data_transfer_emissions_mt_co2e_per_month = data_transfer_emissions_mt_co2e_per_month
+        self.corporate_allocation_percentage = corporate_allocation_percentage
+        self.server_allocation_percentage = server_allocation_percentage
+        self.distribution_partners = distribution_partners
+        self.ad_tech_platform_bid_requests_processed_billion_per_month = (
+            ad_tech_platform_bid_requests_processed_billion_per_month
+        )
+        self.cookie_syncs_processed_billion_per_month = cookie_syncs_processed_billion_per_month
+
+    def get_bid_requests_processed_per_month(self) -> float:
+        return self.bid_requests_processed_billion_per_month * 1000000000
+
+    def get_server_emissions_g_co2e_per_month(self) -> float:
+        return self.server_emissions_mt_co2e_per_month * 1000000
+
+    def get_servers_processing_bid_requests_pct(self) -> float:
```

Comment:
nit but what do you think about reserving `pct` for when it's based on 100, and `rate` for when it's based on 1?

Reply:
👍